test/Npgsql.Tests/CommandBuilderTests.cs

```diff
@@ -812,5 +812,38 @@ public void GetUpdateCommandInfersParametersWithNpgsqDbType()
                 Assert.That(daDataAdapter.Update(dtTable), Is.EqualTo(1));
             }
         }
+
+        [Test, IssueLink("https://github.com/npgsql/npgsql/issues/2560")]
+        public void GetUpdateCommandWithColumnAliases()
+        {
+            using (var conn = OpenConnection())
+            {
+                conn.ExecuteNonQuery(@"
+                    CREATE TABLE pg_temp.test (
+                        Cod varchar(5) NOT NULL,
+                        Descr varchar(40),
+                        Data date,
+                        CONSTRAINT PK_test_Cod PRIMARY KEY (Cod)
+                    );
+                ");
+                var cmd = new NpgsqlCommand("SELECT Cod as CodAlias, Descr as DescrAlias, Data as DataAlias FROM test", conn);
+
+                var daDataAdapter = new NpgsqlDataAdapter(cmd);
+                var cbCommandBuilder = new NpgsqlCommandBuilder(daDataAdapter);
+                daDataAdapter.UpdateCommand = cbCommandBuilder.GetUpdateCommand();
+                Assert.True(daDataAdapter.UpdateCommand.CommandText.Contains("SET \"cod\" = @p1, \"descr\" = @p2, \"data\" = @p3 WHERE ((\"cod\" = @p4) AND ((@p5 = 1 AND \"descr\" IS NULL) OR (\"descr\" = @p6)) AND ((@p7 = 1 AND \"data\" IS NULL) OR (\"data\" = @p8)))"));
+
+                // specify command behaviour of SchemaOnly | KeyInfo for datareader in order for column meta data to be retrieved
+                var dr = cmd.ExecuteReader(CommandBehavior.SchemaOnly|CommandBehavior.KeyInfo);
```

Comment:
Thanks for this testing - it's great. However, can you please move this into a new test in ReaderOldSchemaTests, which specifically tests BaseColumnName? This way, this test makes sure the UpdateCommand is generated correctly, and in ReaderOldSchemaTests we make sure the correct schema info is returned. The same test should also exist in ReaderNewSchemaTests with the newer API.

Reply:
Thanks @roji , will move test and fix nit

Reply:
@roji , i've moved the explicit BaseColumnName tests into a new test called BaseColumnName in the ReaderOldSchemaTests and created a new BaseColumnNameWithColumnAliases test in the ReaderNewSchemaTests (this is the same test as in the ReaderOdSchemaTest but with the newer API as suggested. I thought this was better then modifying the existing BaseColumnName test).
I've also used the new using declaration syntax for the new tests and updated the commandBuilder tests using statements (thanks for the tip, i didn't know about the new syntax. I really like this)